gitlab-runner/src/client.rs

```diff
@@ -275,20 +282,72 @@ pub(crate) struct Client {
     client: reqwest::Client,
     url: Url,
     token: String,
+    system_id: String,
 }
 
 impl Client {
     pub fn new(url: Url, token: String) -> Self {
+        let system_id = Self::generate_system_id();
         Self {
             client: reqwest::Client::new(),
             url,
             token,
+            system_id,
+        }
+    }
+
+    fn generate_system_id_from_machine_id() -> Option<String> {
+        // Ideally this would be async for consistency, but that's for the next API bump really. In
+        // practise the client will be created at the start of the runner and the amount read is
+        // really tiny so blocking is not a real issue.
+        let mut f = match File::open("/etc/machine-id") {
```

Comment:
Worth noting the `machineid` module the runner uses also checks `/var/lib/dbus/machine-id` first, but I'm not sure if that's particularly relevant on a modern system / container.

Reply:
Yeah i don't care abut systems without `/etc/machine-id` really :)